src/utils/components/DiskModal/components/StorageProfileSettings/AccessMode.tsx

```diff
@@ -1,56 +1,56 @@
-import React, { FC, useEffect, useMemo } from 'react';
+import React, { FC } from 'react';
 import { useFormContext } from 'react-hook-form';
 
-import { V1beta1StorageSpecVolumeModeEnum } from '@kubevirt-ui/kubevirt-api/kubevirt';
 import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { ClaimPropertySets } from '@kubevirt-utils/types/storage';
 import { FormGroup, Radio } from '@patternfly/react-core';
 
 import { V1DiskFormState } from '../../utils/types';
-import {
-  ACCESS_MODE_FIELD,
-  ACCESS_MODE_FIELDID,
-  STORAGE_CLASS_PROVIDER_FIELD,
-  VOLUME_MODE_FIELD,
-} from '../utils/constants';
-import { ACCESS_MODE_RADIO_OPTIONS, getAccessModeForProvisioner } from '../utils/modesMapping';
-
-const AccessMode: FC = () => {
+import { ACCESS_MODE_FIELD, ACCESS_MODE_FIELDID, VOLUME_MODE_FIELD } from '../utils/constants';
+import { ACCESS_MODE_RADIO_OPTIONS, getAccessModesForVolume } from '../utils/modesMapping';
+
+import RecommendationLabel from './RecommendationLabel';
+
+import './ApplyStorageProfileSettings.scss';
+
+type AccessModeProps = {
+  claimPropertySets: ClaimPropertySets;
+};
+
+const AccessMode: FC<AccessModeProps> = ({ claimPropertySets }) => {
   const { t } = useKubevirtTranslation();
 
   const { setValue, watch } = useFormContext<V1DiskFormState>();
 
-  const [storageClassProvisioner, accessModes, volumeMode] = watch([
-    STORAGE_CLASS_PROVIDER_FIELD,
-    ACCESS_MODE_FIELD,
-    VOLUME_MODE_FIELD,
-  ]);
+  const [accessModes, volumeMode] = watch([ACCESS_MODE_FIELD, VOLUME_MODE_FIELD]);
```

Comment:
better to useWatch hook, it will cause less rendering then watch()

Reply:
> where do u use setValue before initial state?
>
Already pointed above:
>However we do use setValue before registering the hook i.e. when setting [access mode based on volume mode](https://github.com/rszwajko/kubevirt-plugin/blob/bc4cdcf03cd198b7aeca7820c60b76e882bb407f/src/utils/components/DiskModal/components/StorageProfileSettings/VolumeMode.tsx#L56) or when r[esetting the form values](https://github.com/rszwajko/kubevirt-plugin/blob/bc4cdcf03cd198b7aeca7820c60b76e882bb407f/src/utils/components/DiskModal/components/StorageClassAndPreallocation/StorageClassSelect.tsx#L53) after storage class change
>

> if u have effect that setValue, before all initialize just give a default value
>
There is no static default here. The value needs to be retrieved dynamically.